nixops/storage/legacy.py

```diff
@@ -0,0 +1,54 @@
+from nixops.storage import StorageArgDescriptions, StorageArgValues
+import nixops.statefile
+import sys
+import os
+import os.path
+
+class LegacyBackend:
+    @staticmethod
+    def arguments() -> StorageArgDescriptions:
+        raise NotImplementedError
+
+    def __init__(self, args: StorageArgValues) -> None:
+        pass
+
+    # fetchToFile: acquire a lock and download the state file to
+    # the local disk. Note: no arguments will be passed over kwargs.
+    # Making it part of the type definition allows adding new
+    # arguments later.
+    def fetchToFile(self, path: str, **kwargs) -> None:
+        os.symlink(os.path.abspath(self.state_location()), path)
```

Comment:
This at one point did not abspath, which was causing  problems with legacy paths to state files.

Reply:
It seems to me like it would be a bit safer to make an actual copy rather than a symlink.

That way we know that if something goes wrong in the middle of writing the state file to disk, the "proper"  statefile won't be corrupted.

Reply:
If we were to copy, we would need to implement our own implementation of locking on top. The legacy backend already supports its own locking. My thinking is it would be better to find a way to "escape" the legacy backend, or implement a "backup" feature in the Legacy backend, than implement copying. What do you think?